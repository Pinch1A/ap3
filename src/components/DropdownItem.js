import React from 'react'
import styled from 'styled-components'
import { Flag } from './Dropdown'

const Flex = styled.div`
  display: flex;
`

const Item = styled(Flex)`
  padding: 15px 30px;
  justify-content: flex-start;
  align-items: center;
  cursor: pointer;
  background: ${(props) => (props.isSelected ? '#f5f5f5' : '#fff')};
  color: ${(props) => (props.isSelected ? '#1300cb' : '#000')};
  &:hover {
    background: #f5f5f5;
  }
`
Item.displayName = 'Item'

const DropdownItem = ({ name, flag, isSelected, setSelected }) => {
  return (
    <Item onClick={() => setSelected({ name, flag })} isSelected={isSelected}>
      <Flag src={flag} alt={name} />
      <Flex>{name}</Flex>
    </Item>
  )
}

export default DropdownItem
