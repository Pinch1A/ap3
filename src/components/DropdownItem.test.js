import DropdownItem from './DropdownItem'
import React from 'react'
import { shallow, mount } from 'enzyme'

const props = {
  name: 'Select',
  flag: 'test',
  isSelected: false,
  setSelected: () => jest.fn(),
}

describe('<DropdownItem>', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<DropdownItem {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('setSelected function is called on item click', () => {
    const mockSetSelected = jest.fn()

    const props = {
      name: 'test',
      flag: 'test-flag',
      isSelected: false,
      setSelected: mockSetSelected,
    }

    const wrapper = mount(<DropdownItem {...props} />)
    const item = wrapper.find('Item')
    item.simulate('click')
    expect(props.setSelected.mock.calls.length).toEqual(1)
  })
})
