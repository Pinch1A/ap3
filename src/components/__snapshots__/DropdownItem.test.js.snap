// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownItem> renders without crashing 1`] = `
<Item
  isSelected={false}
  onClick={[Function]}
>
  <styled.img
    alt="Select"
    src="test"
  />
  <styled.div>
    Select
  </styled.div>
</Item>
`;
